### PR TITLE
Fix mood and emotion graph visualization bugs.

### PR DESCRIPTION
**Pull request by eschlange**
* Updated the javascript and mood/emotion data collection logic to solve viz bug.
* Moved mood and emotion visualization logic out of participant and into a new service.
* Updated references of visualization logic in existing controller and view logic.

[#119306809]

### CODE REVIEW

File: app/support/mood_and_emotion_visualization_service.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        }
+        averaged_ratings << daily_negative
+      end
+    end
+    averaged_ratings
+  end
+
+  private
+
+  def average_rating(array)
```
Comment (ericcf) [inline app/support/mood_and_emotion_visualization_service.rb]: Won't this raise a `ZeroDivisionError` if it's an empty array?

Comment (eschlange) [inline app/support/mood_and_emotion_visualization_service.rb]: Yes, but that method never runs if the array has a size of 0.

File: app/assets/javascripts/think_feel_do_engine/feel/draw_graphs.js [hunk 2]
```diff
@@ -25,6 +25,9 @@
         .call(columnChart(graphParameters.startDate, graphParameters.endDate, Y_MIN, Y_MAX, GRAPH_TITLE, YLABEL)
           .width(graphParameters.graphWidth)
           .height(PIXEL_HEIGHT)
+          .x(function(date, i) {
+            return moment(date.day, 'YYYY-MM-DD').startOf('day')._d;
```
Comment (ericcf) [inline app/assets/javascripts/think_feel_do_engine/feel/draw_graphs.js]: Why are you accessing `_d` here? It seems risky to depend on a "private" property.

Comment (eschlange) [inline app/assets/javascripts/think_feel_do_engine/feel/draw_graphs.js]: Mike, do you know the original purpose of accessing ```_d``` in this code?

Comment (michaelwehrley) [inline app/assets/javascripts/think_feel_do_engine/feel/draw_graphs.js]: makes sense; let's not use a private property.  Good catch ⚾️

Comment (eschlange) [inline app/assets/javascripts/think_feel_do_engine/feel/draw_graphs.js]: I found the Moment library method specifically designed to return the wrapped javascript Date object. I updated the code to use the method.

**Comment by ericcf** [comment]
I think the remaining item is testing. Are these graphs covered somehow in acceptance testing @cbrenner04 ?

**Comment by michaelwehrley** [comment]
Thanks for moving the code out of the participant class and creating a service.  My only concern would be the lack of specs for this service.  It seems however through conversation with @eschlange that this may be too tedious and unnecessary given we may have additional feature specs via @cbrenner04.

**Comment by michaelwehrley** [comment]
👍

**Comment by ericcf** [comment]
If @cbrenner04 feels that the acceptance tests give us adequate confidence then I agree tests might be unnecessary. I'm not sure what tediousness has to do with it.

**Comment by cbrenner04** [comment]
Sorry for taking so long to respond. I clearly need to update the acceptance tests as this slipped through them before but otherwise I think I can cover it.

**Pull request merged**